hw/top_darjeeling/templates/toplevel.sv.tpl

```diff
@@ -491,20 +499,41 @@ max_intrwidth = (max(len(x.name) for x in block.interrupts)
   ${m["type"]} #(
 <%include file="/toplevel_racl.tpl" args="m=m,top=top"/>\
   % if block.alerts:
-<%
-w = len(block.alerts)
-if 'outgoing_alert' in m:
-  outgoing_alert = m['outgoing_alert']
-  lo = outgoing_alert_idx[outgoing_alert]
+  <%
+## Assemble array of block alerts
+alert_handlers_used = set(m["alert_connections"].values())
+if len(alert_handlers_used) == 1: ## they're all the same
+  handler, = alert_handlers_used ## get the one handler out of the set
+  w = len(block.alerts)
+  if 'outgoing_alert' in m:
+    outgoing_alert = m['outgoing_alert']
+    lo = outgoing_alert_idx[outgoing_alert]
+    async_exp = "AsyncOnOutgoingAlert" + alert_group.capitalize()
+  else:
+    lo = alert_idx[handler]
+    async_exp = f"{handler}_reg_pkg::AsyncOn"
+  slice = f"{lo+w-1}:{lo}"
+  async_expr = async_exp + "[" + slice + "]"
+  alert_tx_expr = "alert" + handler.replace("alert_handler", "") + f"_tx[{slice}]"
```

Comment:
the "alert_handler" replacement string should really be defined only at one place. I suggested above moving that out to a function.